docs/porting/connectivity/Layer3IP.md

```diff
@@ -0,0 +1,144 @@
+<h2 id="ethernet-port">IP Layer3 (L3IP) drivers</h2>
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/l3ip.png)<span>L3IP API</span></span>
+
+This document describes how to port and test a cellular IP Layer3 (L3IP) driver to Mbed OS.
+
+The scope of this document is limited to advanced cellular modems, where the device presents an additional IP Layer3  interface to send and receive frames, and one of the onboard network stacks that runs on Mbed OS on the host processor uses this.
+
+(If the device has an off-board network stack, a driver needs to implement `NetworkStack` directly instead to pass network calls to the offboard stack).
+
+### Abstractions
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/l3ip-driver.png)<span>L3IP driver</span></span>
+
+The L3IP interface abstracts network stacks and drivers and easily permits multiple instances. The key API classes are:
+
+- `NetworkInterface` - an Mbed OS network interface of any type.
+- `NetworkStack` - an Mbed OS network stack of any type (may be off-board).
+- `OnboardNetworkStack` - an onboard network stack.
+- `L3IP` - an IP Layer3  cellular device driver.
+- `NetStackMemoryManager` - a memory manager used to pass data between driver and stack.
+- `L3IPInterface`- a `NetworkInterface` that uses an `L3IP` driver and an `OnboardNetworkStack`.
+
+### The L3IP driver core
+
+The first step in the port is to create a driver class for cellular device that you can instantiate to control your device. You must derive this class from the `L3IP` class. A network stack (or test framework) uses this API to control your driver.
+
+The L3IP-derived driver would normally be installed in features/cellular/framework/targets, often in a `TARGET_XXX` directory.
```

Comment:
Remove this line, as the porting guide does not need to specify where to place the driver. We allow some flexibility there.

Reply:
done